Runtime/Highlight/ObjectHighlighter.cs

```diff
@@ -0,0 +1,453 @@
+﻿using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering;
+using Innoactive.Creator.Unity;
+
+namespace Innoactive.Creator.BasicInteraction
+{
+    /// <summary>
+    /// Adds highlighting functionality to a GameObject with Renderers.
+    /// </summary>
+    /// <remarks>
+    /// Highlights are always queued following a LIFO (Last In First Out) scheme. 
+    /// </remarks>
+    public class ObjectHighlighter : MonoBehaviour
+    {
+        private readonly Dictionary<string, Material> highlightIDs = new Dictionary<string, Material>();
+
+        protected bool IsBeingHighlighted => highlightIDs.Any();
+
+        [SerializeField, HideInInspector]
+        private Renderer[] renderers = {};
+        
+        [SerializeField, HideInInspector]
+        private MeshRenderer highlightMeshRenderer = null;
+        
+        [SerializeField , HideInInspector]
+        private MeshFilter highlightMeshFilter = null;
+        
+        protected virtual void Reset()
+        {
+            RefreshCachedRenderers();
+        }
+
+        protected virtual void OnDisable()
+        {
+            if (IsBeingHighlighted)
+            {
+                ReenableRenderers();
+                highlightIDs.Clear();
+            }
+        }
+
+        /// <summary>
+        /// Returns true if there is any highlight active.
+        /// </summary>
+        /// <param name="listOfActiveHighlights">In case there is any active highlight, this value will be filled with all available highlight IDs.</param>
+        public bool IsHighlighting(out IEnumerable<string> listOfActiveHighlights)
+        {
+            listOfActiveHighlights = null;
+            List<string> listOfHighlights = new List<string>();
+
+            if (IsBeingHighlighted)
+            {
+                foreach (KeyValuePair<string, Material> highlightID in highlightIDs)
+                {
+                    listOfHighlights.Add(highlightID.Key);
+                }
+
+                listOfActiveHighlights = listOfHighlights;
+            }
+
+            return IsBeingHighlighted;
+        }
+
+        /// <summary>
+        /// Returns true if there is any highlight with provided <paramref name="highlightID"/>.
+        /// </summary>
+        /// <remarks>
+        /// A <paramref name="highlightID"/> can be highlighting but not visible due to another highlight on course.
+        /// </remarks>
+        /// <param name="highlightID"> ID for the highlighting.</param>
+        /// <param name="isCurrentlyVisible"> True if the <paramref name="highlightID"/> is being highlighted and is currently visible.</param>
+        public bool IsHighlighting(string highlightID, out bool isCurrentlyVisible)
```

Comment:
why are you combining two methods in one with an odd out thingi, not good.

Reply:
Same as before:

> A highlighter can highlight multiple times the same object, only the last requested highlight is shown, this method intends to tell if there is any highlight going on and to provide a list of which highlights are happening at the same time.
> 
> I can split it if you think is cleaner that way.

So basically you want to know if the highlight from the highlight behavior is going on if the highlighter is being grabbed this method would only tell you that is being highlighted but you don't know by who.